### Курс «Теория вероятностей и математическая статистика»
### Практическое задание к уроку 2
## Тема «Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона»

__Задача 1__

Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании. Найдите `M(X), D(X)`.

Вероятность случайного угадывания правильного ответа в отдельно взятом вопросе равна:

### $$p=\frac{m}{n}=\frac{1}{4}=0.25$$

Посчитаем распределение числа угадывания правильного ответа при пяти вопросах в контрольной работе. Возможные значения такой случайной величины:  

$𝑥_1=0,  𝑥_2=1,  𝑥_3=2,  𝑥_4=3, 𝑥_5=4, 𝑥_6=5.$

Посчитаем вероятность каждого значения:

In [1]:
import numpy as np

In [2]:
def combinations(n, k):
    return int(np.math.factorial(n) / (np.math.factorial(k) * np.math.factorial(n - k)))

def binominal(n, k, p):
    return combinations(n, k) * (p ** k) * ((1 - p) ** (n - k))

# Количество вопросов
n = 5

# Вероятность угадывания правильного ответа из четырех
p1 = 1 / 4

In [3]:
# Закон распределения
p = []

for k in range(n + 1):
    p.append(binominal(n, k, p1))
p

[0.2373046875,
 0.3955078125,
 0.263671875,
 0.087890625,
 0.0146484375,
 0.0009765625]

Закон распределения:

<table border="3">
    <tr>
        <th>$x$</th>
        <td>0</td>
        <td>1</td>
        <td>2</td>
        <td>3</td>
        <td>4</td>
        <td>5</td>
    </tr>
    <tr>
        <th>$P(X = x)$</th>
        <td>0.23730</td>
        <td>0.39551</td>
        <td>0.26367</td>
        <td>0.08789</td>
        <td>0.01465</td>        
        <td>0.00098</td>                
    </tr>
</table>

Посчитаем математическое ожидание распределения. Поскольку $n = 5$ и $p = 0.25$, получаем:

$$M(X) = 5 \cdot 0.25 = 1.25$$

Дисперсия такой величины:

$$D(X) = 5 \cdot 0.25 \cdot (1 - 0.25) = 0.9375$$

__Задача 2__

Вероятность поймать рыбу при одном забрасывании удочки равна 0.01. Какова вероятность того, что число пойманных рыб при 200 забросах удочки будет не менее 5 и не более 10?

$a = 200 \cdot 0.01 = 2$

$k\in[5,10]$

In [4]:
def pois(a, k):
    return (a ** k) * (np.exp(-a)) / np.math.factorial(k)

In [5]:
a = 2

# Сумма вероятностей того, что число пойманных рыб 
# будет не менее 5 и не более 10
sum(pois(a=a,k=i) for i in range(5,11))

0.052644709119342686

__Задача 3__

В условиях предыдущей задачи сколько забросов удочки нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 10 раз будет поймана рыба?

_Подсказка_. Как посчитать `P(X >= k)` для величины `X`, имеющей биномиальное распределение? Здесь предстоит немножко покодить.

In [6]:
# Требуемая вероятность
p_u = 0.9
# Требуемое число пойманных рыб
k = 10
# Вероятность поймать рыбу
p_n = 0.01
# Данные для инициализаци
p = 1
n = 0

while p > (1 - p_u):
    n += 1
    a = n * p_n
    p = sum(pois(a=a,k=i) for i in range(k))

print(n,(1 - p))

1421 0.9001760722984115
